qemu/tests/qemu_disk_img_snapshot.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from autotest.client.shared import error
+from qemu.tests import qemu_disk_img
+
+
+class SnapshotTest(qemu_disk_img.QemuImgTest):
+
+    def __init__(self, test, params, env):
+        self.tag = params.get("image_snapshot", "image1")
+        t_params = params.object_params(self.tag)
+        super(SnapshotTest, self).__init__(test, t_params, env, self.tag)
+
+
+def run(test, params, env):
+    """
+    'qemu-img' snapshot functions test:
+
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    base_image = params.get("images", "image1").split()[0]
+    params.update(
+        {"image_name_%s" % base_image: params["image_name"],
+         "image_format_%s" % base_image: params["image_format"]})
+    t_file = params["guest_file_name"]
+    snapshot_test = SnapshotTest(test, params, env)
+
```

Comment:
Please print your test steps use error.context("step xxxx", logging.info), so that tester can catch your testing via read INFO log.